src/main/java/com/laytonsmith/core/functions/ArrayHandling.java

```diff
@@ -144,21 +142,14 @@ public Construct exec(Target t, Environment env, Construct... args) throws Cance
 			if (args[0] instanceof CArray) {
 				CArray ca = (CArray) args[0];
 				if (index instanceof CSlice) {
-					if (ca.inAssociativeMode()) {
-						if (((CSlice) index).getStart() == 0 && ((CSlice) index).getFinish() == -1) {
-							//Special exception, we want to clone the whole array
-							CArray na = CArray.GetAssociativeArray(t);
-							for (Construct key : ca.keySet()) {
-								try {
-									na.set(key, ca.get(key, t).clone(), t);
-								} catch (CloneNotSupportedException ex) {
-									na.set(key, ca.get(key, t), t);
-								}
-							}
-							return na;
-						}
+						
+					// Deep clone the array if the "index" is the initial one.
+					if (((CSlice) index).getStart() == 0 && ((CSlice) index).getFinish() == -1) {
```

Comment:
Remove this case. I will make it so that @a[] calls array_deep_clone instead, so this will only be triggered when the user is doing a slice, which will remain a shallow copy.